core/bin/external_node/src/main.rs

```diff
@@ -424,19 +430,41 @@ async fn run_api(
             )
         });
 
+        let tokens_whitelisted_for_paymaster_cache = Arc::new(RwLock::new(Vec::new()));
+        let tokens_whitelisted_for_paymaster_cache_clone =
+            tokens_whitelisted_for_paymaster_cache.clone();
+        let whitelisted_tokens_update_task = task::spawn(async move {
+            loop {
+                match main_node_client.tokens_whitelisted_for_paymaster().await {
+                    Ok(tokens) => {
+                        *tokens_whitelisted_for_paymaster_cache_clone.write().await = tokens;
+                    }
+                    Err(err) => {
+                        tracing::error!(
+                            "Failed to query `tokens_whitelisted_for_paymaster`, error: {err:?}"
+                        );
+                    }
+                }
+
+                tokio::time::sleep(Duration::from_secs(60)).await;
```

Comment:
Please make the task cancel-aware [like here](https://github.com/matter-labs/zksync-era/blob/4f803711da6df5e75ef2c08a42b0029a9a84969a/core/bin/external_node/src/main.rs#L518) so that the EN doesn't hang up on it during shutdown.